### PR TITLE
adds support for setting a timeout per packet

### PR DESCRIPTION
**Pull request by ivokanchev**

### CODE REVIEW

File: ping.go [hunk 1]
```diff
@@ -476,16 +477,36 @@ func (p *Pinger) runLoop(
...
 	}()
 
 	if err := p.sendICMP(conn); err != nil {
 		return err
 	}
 
+	go func() {
+		for {
+			select {
+			case <-packetTimeout.C:
```
Comment (simeonkorchev) [inline ping.go]: The select has only 1 case, so does it make sense to have for range over packetTimeout.C?

Comment (simeonkorchev) [inline ping.go]: The select has only 1 case, so does it make sense to have for range over packetTimeout.C?

File: ping.go [hunk 2]
```diff
@@ -512,9 +533,32 @@ func (p *Pinger) runLoop(
...
+		p.awaitingSequencesLock.RLock()
+		if p.Count > 0 && p.PacketsRecv+p.PacketsTimeoutExceeded >= p.Count {
 			return nil
 		}
+		p.awaitingSequencesLock.RUnlock()
+	}
+}
+
+func (p *Pinger) checkInFlightPackets() {
+	// Loop through each item in map
```
Comment (simeonkorchev) [inline ping.go]: nit: not needed comment.

**Review by simeonkorchev — APPROVED** [review]
Overall looks good. I only left a few not so important comments.
Still,  I think that with the current implementation it is very easy to forget to add the needed locks, when dealing with awaitingSequences. Extracting the awaitingSequence and its lock into a separate struct, safe to use from multiple go routines would improve this situation and the clients won't bother dealing with locks.

**Comment by ivokanchev** [comment]
Code is refactored using a simpler approach to achieve the same result.

**Pull request merged**